management/server/account.go

```diff
@@ -1345,31 +1349,38 @@ func (am *DefaultAccountManager) getAccountFromCache(accountID string, forceRelo
 	}
 }
 
-func (am *DefaultAccountManager) lookupCache(accountUsers map[string]struct{}, accountID string) ([]*idp.UserData, error) {
+func (am *DefaultAccountManager) lookupCache(accountUsers map[string]userLoggedInOnce, accountID string) ([]*idp.UserData, error) {
 	data, err := am.getAccountFromCache(accountID, false)
 	if err != nil {
 		return nil, err
 	}
 
-	userDataMap := make(map[string]struct{})
+	userDataMap := make(map[string]*idp.UserData, len(data))
 	for _, datum := range data {
-		userDataMap[datum.ID] = struct{}{}
+		userDataMap[datum.ID] = datum
 	}
 
+	mustRefreshInviteStatus := false
+
 	// the accountUsers ID list of non integration users from store, we check if cache has all of them
 	// as result of for loop knownUsersCount will have number of users are not presented in the cashed
 	knownUsersCount := len(accountUsers)
-	for user := range accountUsers {
-		if _, ok := userDataMap[user]; ok {
+	for user, loggedInOnce := range accountUsers {
+		if datum, ok := userDataMap[user]; ok {
+			// check if the matching user data has a pending invite and if the user has logged in once, forcing the cache to be refreshed
+			if datum.AppMetadata.WTPendingInvite != nil && *datum.AppMetadata.WTPendingInvite && loggedInOnce == true { //nolint:gosimple
+				mustRefreshInviteStatus = true
+				break
+			}
 			knownUsersCount--
 			continue
 		}
 		log.Debugf("cache doesn't know about %s user", user)
 	}
 
 	// if we know users that are not yet in cache more likely cache is outdated
-	if knownUsersCount > 0 {
-		log.Debugf("cache doesn't know about %d users from store, reloading", knownUsersCount)
+	if knownUsersCount > 0 && mustRefreshInviteStatus {
+		log.Debugf("reloading cache with IDP manager. Users unkown to the cache: %d, Must refresh invite status? %t", knownUsersCount, mustRefreshInviteStatus)
```

Comment:
Shouldn't this be an OR operator? otherwise, either `mustRefreshInviteStatus` or `knownUsersCount` is unnecessary as the break will make sure it is > 0. Also the break will falsify the log message message becasue `knownUsersCount` doesn't finish the loop